yateto/ast/cost.py

```diff
@@ -71,14 +72,50 @@ def estimate_Product(self, node):
     self._cache[node] = bb
 
     return bb.size()
-  
+
   def estimate_IndexSum(self, node):
     tbb = self._cache[node.term()]
     pos = node.term().indices.find(str(node.sumIndex()))
     bb = BoundingBox([r for i,r in enumerate(tbb) if i != pos])
     self._cache[node] = bb
     return tbb.size() - bb.size()
 
+
+class GpuBoundingBoxCostEstimator(BoundingBoxCostEstimator):
+  """Estimates num. of hardware flops for a tensor operation per GPU thread.
+  Therefore, results of BoundingBoxCostEstimator are divided by a size
+  of the first dimension of lhs because this dimension is fully parallelized.
+  Note, the estimator includes GPU caching.
+  """
+  def __init__(self):
+    super().__init__()
+    self._lid_dim = 0
+    self._loaded_to_gpu_cache = set()
```

Comment:
Could you explain this? These estimators are only about flops and not caching effects

Reply:
see above